include/unifex/async_scope.hpp

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/async_manual_reset_event.hpp>
+#include <unifex/get_stop_token.hpp>
+#include <unifex/inplace_stop_token.hpp>
+#include <unifex/just.hpp>
+#include <unifex/manual_lifetime.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/scheduler_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/sequence.hpp>
+#include <unifex/transform.hpp>
+#include <unifex/type_traits.hpp>
+
+#include <atomic>
+#include <memory>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+namespace _async_scope {
+
+struct async_scope;
+
+struct _receiver_base {
+  [[noreturn]] void set_error(std::exception_ptr) noexcept {
+    std::terminate();
+  }
```

Comment:
Perhaps what I need is more documentation, but the intent here is that the submitted work needs to handle errors rather than let them bubble up to the `async_scope`.  The given _Sender_ needs to either complete with `set_value()` or `set_done()`.  I _wanted_ to define the internal _Receiver_ without a `set_error` at all so that _Senders_ that _try_ to complete with an error would just not compile, but a type without a `set_error` doesn't match the `receiver` concept so  I had to take a different approach.

Reply:
>  type without a `set_error` doesn't match the `receiver` concept so  I had to take a different approach

This is one of the reasons I would like to see the receiver concept broken up into separate `error_receiver<E>`, `value_receiver<Ts...>`, `done_receiver` concepts rather than having the base receiver concept always require set_error/set_done.